### ISSUE TITLE
Format table get unexpected right aligment

### ISSUE THREAD

**Issue by riyueniao** [AH-Hammi/pandoc-grid-table#1]
My input grid table.

```
+-----+-------------------------+
| No. | Description             |
+:===:+=========================+
| 1   | - This is a long string |
|     |   + Short string        |
|     |   + Short string        |
|     |   + Short string        |
|     |   + Short string        |
|     |   + Short string        |
+-----+-------------------------+
```

After pressing `TAB`. formatted table as below. which is not the format I expected, and will cause the list not working.
```
+-----+-------------------------+
| No. | Description             |
+:===:+=========================+
| 1   | - This is a long string |
|     |          + Short string |
|     |          + Short string |
|     |          + Short string |
|     |          + Short string |
|     |          + Short string |
+-----+-------------------------+
```

**Comment by AH-Hammi** [comment]
Hey Preston thanks for sharing this issue with me I will look into it once I have time again

**Comment by AH-Hammi** [comment]
Hey Preston, this issue comes due to a missing check. It is only right aligned due to the "+" char which indicates a new cell. I think I there should be a check if we have a valid cell separator above or something.

Thanks for opening

**Comment by AH-Hammi** [comment]
Hey Preston, this issue comes due to a missing check. It is only right aligned due to the "+" char which indicates a new cell. I think I there should be a check if we have a valid cell separator above or something.

Thanks for opening

**Comment by AH-Hammi** [comment]
Hey Preston, did you implement a test for the new case and gone ahead and tested your code?
I want to have 100% Coverage on all scripts except the extension.ts file. Just run `npm run coverage and take a look at the coverage report.

And maybe add a test with your example with the expected result.

Afterwards you can open a pull request. Thanks for your participation 😄

**Comment by riyueniao** [comment]
Looks like `|` and `+`  in a table cell are special cases.
Which will cause the below code to create incorrect table cells.
```
if (char === "|" || char === "+") {
    // create a new cell with the current cell string
    this._cells.push(new TableCell(current_cell_str));
    // reset the current cell string
    current_cell_str = char;
}
```

**Comment by riyueniao** [comment]
Hi, I have fixed the issue by checking the table cell is valid using a regex.
Following is the coverage result.

```
-----------------|---------|----------|---------|---------|--------------------------------------------------------------------------------------
File             | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
-----------------|---------|----------|---------|---------|--------------------------------------------------------------------------------------
All files        |   56.23 |    96.24 |   73.86 |   56.23 |
 commands.ts     |     100 |      100 |     100 |     100 |
 commands.ts     |    9.67 |      100 |    12.5 |    9.67 | 22-68,71-105,107-150,152-200,202-259,262-288,291-310
 extension.ts    |     100 |      100 |     100 |     100 |
 table.ts        |     100 |      100 |     100 |     100 |
 table.ts        |    6.19 |        0 |       0 |    6.19 | 15-82,85-193,196-252,255-323
 table_cell.ts   |     100 |      100 |     100 |     100 |
 table_cell.ts   |    4.54 |       50 |       0 |    4.54 | 6-46,49-154
 table_column.ts |     100 |      100 |     100 |     100 |
 table_column.ts |   18.42 |        0 |       0 |   18.42 | 8-38
 table_row.ts    |     100 |      100 |     100 |     100 |
 table_row.ts    |    3.89 |        0 |       0 |    3.89 | 4-77
 text_utility.ts |     100 |      100 |     100 |     100 |
 text_utility.ts |   23.64 |       80 |   11.11 |   23.64 | 15-16,32-34,50-52,63-65,77-87,89-109,111-142,144-170,173-226,229-282,284-292,294-313
-----------------|---------|----------|---------|---------|--------------------------------------------------------------------------------------
```

**Comment by AH-Hammi** [comment]
Could you fork the repository to your own GitHub account and push the changes there?
Then I will be able to review your code and check what you did, because I would really like to understand what you did with regex :smile:

**Comment by riyueniao** [comment]
You check it by [diff.log](https://github.com/user-attachments/files/16106420/diff.log)

**Comment by AH-Hammi** [comment]
Hey @riyueniao, the test cases you supplied have a malformed table.
The examples work without your changes to the code. Therefore, I will reject those changes.
Yes, “+” and “|” are special characters that need to handled separately.
A Row separator is marked with + left and right to it, a table corner.
This is important, because if inside of two + only valid separator characters are present, either - or = signs.

Your initial Problem doesn't seem to be fixed

**Comment by AH-Hammi** [comment]
I decided to not support + as a valid element inside a grid cell. Therefore, please use another - as a bullet point marker.
Sadly, this also didn't work right now, and I needed to make adjustments to my code.
I created a fix and will publish it as soon as possible.

**Comment by AH-Hammi** [comment]
[Released v0.0.2](https://github.com/AH-Hammi/pandoc-grid-table/releases/tag/v0.0.2)

**Comment by AH-Hammi** [comment]
[Released v0.0.2](https://github.com/AH-Hammi/pandoc-grid-table/releases/tag/v0.0.2) with updated cell trimming.
Won't support "+" inside of cells
